flytekit/core/base_task.py

```diff
@@ -212,6 +214,9 @@ def __init__(
         self._task_type_version = task_type_version
         self._security_ctx = security_ctx
         self._docs = docs
+        self._timeout = timeout
+        # put timeout to metadata here
+        # raise error if metadata and timeout are both set
```

Comment:
I do not like how `timeout` can be defined in two places. If we really want a easy way to set `timeout`, I would do this logic `FileSensor` and not complicate the `Task` API.